### DISCUSSION TITLE
Clone git repo on remote host with ssh key from KeyStore?

### DISCUSSION THREAD

**Discussion by jan-from-meiro** [semaphoreui/semaphore#1191] [Q&A]
Is there a way to clone git repo on remote host with ssh key (or at least token) from KeyStore? Or any other secure way to perform git clone/pull on remote servers without necessity to store credentials there?

**Comment by ansibleguy** [comment]
Greetings.

The documentation says `For remote Repositories that use SSH, you will need to use your SSH key in the Key Store` => https://docs.ansible-semaphore.com/user-guide/repositories#authentication

So I would guess cloning by using ssh-key is supported.

\- AnsibleGuy

**Comment by stafwag** [comment]
I opened a somewhat related discussions:

https://github.com/semaphoreui/semaphore/discussions/1744
https://github.com/semaphoreui/semaphore/discussions/1941

**Comment by fiftin** [comment]
Hi @jan-from-meiro ,

currently Semaphore supports runners and you can start runner on remove server.

Passing secrets to remove server without runner isn't possible because I don't know to secure send secrets from Semaphore to ansible-paybook.
